client/src/entrypoints/images/image-chooser-modified.js

```diff
@@ -0,0 +1,25 @@
+class ImageBlockDefinition extends window.wagtailStreamField.blocks
+  .StructBlockDefinition {
+  render(placeholder, prefix, initialState, initialError) {
+    const block = super.render(placeholder, prefix, initialState, initialError);
+
+    const altTextField = document.getElementById(prefix + '-alt_text');
+    const isDecorativeField = document.getElementById(prefix + '-decorative');
+    const updateStateInput = () => {
+      if (isDecorativeField.checked) {
+        altTextField.setAttribute('disabled', true);
+      } else {
+        altTextField.removeAttribute('disabled');
+      }
```

Comment:
For later – for the rich text implementation, we also add an asterisk next to the alt text field label to convey it’s required. Though this is broken currently.

We should be consistent, either remove the currently-broken rich text code that does this, or implement the pattern in both places.